set.go

```diff
@@ -438,3 +438,38 @@ func (s *Set) registerMetric(name string, m metric) {
 		panic(fmt.Errorf("BUG: metric %q is already registered", name))
 	}
 }
+
+// UnregisterMetric will remove a registered metric
+func (s *Set) UnregisterMetric(name string) {
+	if err := validateMetric(name); err != nil {
+		panic(fmt.Errorf("BUG: invalid metric name %q: %s", name, err))
+	}
+	s.mu.Lock()
+	nm, ok := s.m[name]
+	if ok {
+		delete(s.m, name)
+		for i, a := range s.a {
+			if a == nm {
+				s.a = append(s.a[:i], s.a[i+1:]...)
+			}
+		}
+	}
+	s.mu.Unlock()
+}
+
+// HasMetric will return true, if the metric exists
+func (s *Set) HasMetric(name string) bool {
```

Comment:
Do we really need this method? What is the use case for this method if `UnregisterMetric` would return `bool` as described in the comment above?

Reply:
This is more or less just a helper, I'm not sure it is required. I had a `GetMetric` method first but did not require it in my code as we used `GetOrCreateCounter` and finally removed it. 

So `HasMetric` is used only in tests currently and could be made private to moved to the tests as a helper.

Let me know your thoughts.

Reply:
Let's keep it private if it is needed in tests. Public methods should be introduced with care, since they will remain in the API forever and they cannot be changed due to backwards compatibility.